src/kv/test/kv_snapshot.cpp

```diff
@@ -165,6 +165,83 @@ TEST_CASE("Simple snapshot" * doctest::test_suite("snapshot"))
   }
 }
 
+TEST_CASE("Old snapshots" * doctest::test_suite("snapshot"))
```

Comment:
Added this test for a simpler repro of the incompatibility that was failing the `lts_compat` test, that I alluded to in [this comment](https://github.com/microsoft/CCF/pull/4145/files#r949289405).

Basically the first version of this PR was too aggressive, and lost all knowledge of tombstone entries in serialized snapshots. This meant we couldn't parse snapshots from old nodes, which is a major compatibility break. This test checks we can still parse both kinds of snapshots, and we do that by retaining a few "is this Version negative" checks.